libs/lb/lb-rs/src/rpc.rs

```diff
@@ -0,0 +1,139 @@
+#[derive(Serialize, Deserialize)]
+pub struct RpcRequest {
+    pub method: String,
+    pub args: Option<Vec<u8>>,
+}
+
+#[derive(Serialize, Deserialize)]
+pub enum CallbackMessage<S, T> {
+    Status(S),
+    Done(LbResult<T>),
+}
+
+impl RpcRequest {
+    pub fn new(method: impl Into<String>, args: Option<Vec<u8>>) -> Self {
+        Self {
+            method: method.into(),
+            args,
+        }
+    }
+}
+
+pub async fn call_rpc<T>(
+    socket_address: SocketAddrV4,
+    method: &str,
+    args: Option<Vec<u8>>,
+) -> LbResult<T>
+where
+    T: for<'de> Deserialize<'de>,
+{
+    let mut stream = TcpStream::connect(socket_address)
+            .await
+            .map_err(core_err_unexpected)?;
+    let req = RpcRequest::new(method, args);
+    let encoded = bincode::serialize(&req).map_err(core_err_unexpected)?;
```

Comment:
Here all calls get serialized for the second time. Serialization is generally a pretty expensive operation. It's possible rust / bincode realize that they're serializing a vec (that's already serialized) and do something efficient, but I don't want to have to reason about that.

So I think `RpcRequest` should take a `<T>` of some general form which all the tuples will be pushed into.

This would also remove all the `bincode::serialize` statements from dispatch, further reducing the scope of repeat.

A lot of these wrapper files which list basically every API operation, should be as maintainable as possible. So if we can push complexity that's repeated into a function like this it's a great outcome.